builder/azure/arm/config.go

```diff
@@ -514,16 +514,17 @@ func assertRequiredParametersSet(c *Config, errs *packer.MultiError) {
 	if isUseDeviceLogin(c) {
 		c.useDeviceLogin = true
 	} else {
-		if c.ClientID == "" {
-			errs = packer.MultiErrorAppend(errs, fmt.Errorf("A client_id must be specified"))
+		if (c.ClientID == "" && c.ClientSecret != "") || (c.ClientID != "" && c.ClientSecret == "") {
```

Comment:
NIT: the parenthesis are not needed.

If you introduce a new permutation of flags please create a corresponding test in config_test.go to assert the desired behavior.

Reply:
( making myself a suggestion to see how it works: )
```suggestion
		if c.ClientID == "" && c.ClientSecret != "" || c.ClientID != "" && c.ClientSecret == "" {
```